rules/TypedCollections/Rector/ClassMethod/RemoveNullFromNullableCollectionTypeRector.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Doctrine\TypedCollections\Rector\ClassMethod;
+
+use PhpParser\Node;
+use PhpParser\Node\Name\FullyQualified;
+use PhpParser\Node\NullableType;
+use PhpParser\Node\Stmt\ClassMethod;
+use Rector\Doctrine\Enum\DoctrineClass;
+use Rector\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Doctrine\Tests\TypedCollections\Rector\ClassMethod\RemoveNullFromNullableCollectionTypeRector\RemoveNullFromNullableCollectionTypeRectorTest
+ */
+final class RemoveNullFromNullableCollectionTypeRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Remove null from a nullable Collection, as empty ArrayCollection is preferred instead to keep property type strict and always a collection',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+use Doctrine\Common\Collections\Collection;
+
+final class SomeClass
+{
+    private $items;
+
+    public function setItems(?Collection $items): void
+    {
+        $this->items = $items;
+    }
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+use Doctrine\Common\Collections\Collection;
+
+final class SomeClass
+{
+    private $items;
+
+    public function setItems(Collection $items): void
+    {
+        $this->items = $items;
+    }
+}
+CODE_SAMPLE
+                ),
+
+            ]
+        );
+    }
+
+    public function getNodeTypes(): array
+    {
+        return [ClassMethod::class];
+    }
+
+    /**
+     * @param ClassMethod $node
+     */
+    public function refactor(Node $node): ClassMethod|null
+    {
+        $hasChanged = false;
+
+        foreach ($node->params as $param) {
+            if (! $param->type instanceof NullableType) {
+                continue;
+            }
+
+            $realType = $param->type->type;
+            if (! $this->isName($realType, DoctrineClass::COLLECTION)) {
+                continue;
+            }
+
+            $param->type = new FullyQualified(DoctrineClass::COLLECTION);
```

Comment:
this can just use:

```suggestion
            $param->type = $param->type->type;
```

Reply:
I created PR:

- https://github.com/rectorphp/rector-doctrine/pull/397

for it.